### PR TITLE
Feature/detail fragment

### PR DESCRIPTION
**Pull request by benjivou**

### CODE REVIEW

File: app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt [hunk 1]
```diff
@@ -51,11 +55,17 @@ class MainViewModel : ViewModel() {
 
     val moviesViewHolderListener = object : MovieViewHolder.MoviesViewHolderListener {
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt]: The viewModel should not do this. You have to make your fragment implement the ViewHolder interface the make the actions from the callback in your fragment

File: app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt [hunk 2]
```diff
@@ -51,11 +55,17 @@ class MainViewModel : ViewModel() {
...
+            doOnLikePress(movie)
+        }
+
+        override fun onDetailsRequested(
+            view: View,
+            movie: Movie
+        ) {
+            Log.d(TAG, "onDetailsRequested: image is clicked")
+            val action =
+                MainFragmentDirections.actionMainFragmentToDetailFragment(movie.id.toString())
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt]: Navigation should be done from the fragment, plus, why do you convert your int into a string then in your DetailViewModel convert your string into a int 🧐

File: app/src/main/res/layout/detail_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (alba-benoit) [inline app/src/main/res/layout/detail_fragment.xml]: It is best to use a ConstraintLayout, btw the layout is not very sexy =D

File: app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import com.example.movieapp.ui.fragment.DetailFragment
+
+private const val TAG = "DetailViewModel"
+
+class DetailViewModel : AbstractViewModel(), DetailFragment.DetailFragmentListener {
+
+    private var currentId: MutableLiveData<String?> = MutableLiveData()
+    private var currentMoviePair = MediatorLiveData<Pair<Movie?, Boolean>>()
+    private var movieCurrent = Transformations.switchMap(currentId) {
+        it?.let { internetCall(it) } ?: null
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt]: ?: null is useless

File: app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    private var isLikedMovie: LiveData<Boolean> =
+        Transformations.switchMap(currentId) {
+            Log.i(TAG, "getMovieAndIsLiked: ${it}")
+            it?.toInt()?.let { it1 -> App.database.movieDAO().isLiked(it1) }
+                ?: MutableLiveData()
+        }
+
+    init {
+        currentMoviePair.addSource(movieCurrent) {
+            currentMoviePair.value = Pair(it, isLikedMovie?.value ?: false)
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt]: isLikedMovie is not nullable

File: app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        currentMoviePair.addSource(isLikedMovie) { isLiked ->
+            Log.i(TAG, "isLiked : modif de la list de movie ")
+            currentMoviePair.value =
+                Pair(movieCurrent.value, isLiked)
+        }
+    }
+
+    fun getMovieAndIsLiked(idMovie: String): LiveData<Pair<Movie, Boolean>> {
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt]: why not returning a MediatorLiveData<Pair<Movie?, Boolean>> ?

**Review by alba-benoit — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt (hunk 1 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt]: dasn le fragment implémenter le view holder. adapter passe this

File: app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt (hunk 6 above)
Comment (benjivou) [reply app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt]: Return null 
L'abonnement se fait dans une autre méthode ( _currentMoviePaire)

Comment (benjivou) [reply app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt]: private var _currentMoviePair = MediatorLiveData<Pair<Movie?, Boolean>>()
val currentMoviePair: LiveData<Pair<Movie?, Boolean>>
    get() = _currentMoviePair

File: app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                    )
+                }
+            else
+                _currentMoviePair.value = moviePrepared
+        }
+
+        _currentMoviePair.addSource(isLikedMovie)
+        { isLiked ->
+            Log.d(TAG, "isLikedMLovie : Modified")
+            val toto = _currentMoviePair.value
```
Comment (alba-benoit) [inline app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt]: who is toto ?

**Review by alba-benoit — CHANGES_REQUESTED** [review]

**Review by alba-benoit — APPROVED** [review]

**Pull request merged**